distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso

```diff
@@ -0,0 +1,582 @@
+
+       ! Error Conditions
+
+         - If `s` is incorrectly formatted, a `Number_Parse_Error` is thrown.
+         - If the construction of the Decimal results in a loss of precision, a
+           `Loss_Of_Numeric_Precision` warning is attached. This can only happen
+           if a `Math_Context` value is explicitly passed.
+
+        ^ Example
+          Create a `Decimal` from a string.
+
+              d = Decimal.from_text "12.345"
+    from_text : Text -> Math_Context -> Decimal ! Arithmetic_Error | Number_Parse_Error
+    from_text (s : Text) (mc : Math_Context | Nothing)=Nothing -> Decimal | Number_Parse_Error =
+        handle_java_exception <| handle_number_format_exception <|
+            case mc of
+                _ : Math_Context -> Decimal.Value <| handle_precision_loss s <| Decimal_Utils.fromString s mc.math_context
+                _ : Nothing -> Decimal.Value (Decimal_Utils.fromString s)
+
+    ## GROUP conversions
+       Construct a `Decimal` from an `Integer`.
+
+       Arguments:
+       - i: The `Integer` to construct a `Decimal` from.
+       - mc: The `Math_Context` to use to specify precision and `Rounding_Mode`.
+         If a `Math_Context` is used, there is a possibility of a loss of
+         precision.
+
+       ! Error Conditions
+
+         - If the construction of the Decimal results in a loss of precision, a
+           `Loss_Of_Numeric_Precision` warning is attached. This can only happen
+           if a `Math_Context` value is explicitly passed.
+
+        ^ Example
+          Create a `Decimal` from an integer.
+
+              d = Decimal.from_integer 12
+    from_integer : Integer -> Math_Context -> Decimal ! Arithmetic_Error | Loss_Of_Numeric_Precision
+    from_integer (i : Integer) (mc : Math_Context | Nothing)=Nothing -> Decimal ! Loss_Of_Numeric_Precision =
+        handle_java_exception <|
+            case mc of
+                _ : Math_Context -> Decimal.Value <| handle_precision_loss i <| Decimal_Utils.fromInteger i mc.math_context
+                _ : Nothing -> Decimal.Value (Decimal_Utils.fromInteger i)
+
+    ## GROUP conversions
+       Construct a `Decimal` from a `Float`.
+
+       Arguments:
+       - f: The `Float` to construct a `Decimal` from.
+       - mc: The `Math_Context` to use to specify precision and `Rounding_Mode`.
+         If a `Math_Context` is used, there is a possibility of a loss of
+         precision.
+
+       ? Precision Loss
+
+         A `Decimal` can represent any `Float` precisely. However, some `Float`
+         literals, such as `0.1`, cannot be represented exactly to infinite
+         precision by a `Float`. For this reason, constructing a `Decimal` from
+         a `Float` always attaches a `Loss_Of_Numeric_Precision` warning to the
+         result.
+
+       ! Error Conditions
+
+         - A `Loss_Of_Numeric_Precision` warning is always attached when
```

Comment:
Use of `Loss_Of_Integer_Precision` here might be confusing:

`Decimal.from_float 0.25`

will produce a warning of the form `Loss_Of_Integer_Precision.Error 0.25 0.25`, with two identical values as the error arguments.

Reply:
So we don't use `Loss_Of_Integer_Precision` but `Loss_Of_Numeric_Precision` here after all, right?